src/pages/DelegatoryValidator/Components/TimeDurationIntervalBar.tsx

```diff
@@ -12,37 +14,55 @@ export default function TimeDurationIntervalBar({
     return null;
   }
 
+  const [remainingTime, setRemainingTime] = useState<moment.Duration>();
+  const [percentageComplete, setPercentageComplete] = useState<number>(0);
+
   // the beginning of the unlock cycle
   const startTime = parseTimestamp(timestamp.toString()).subtract(30, "days");
 
   // the end of the unlock cycle
   const unlockTime = parseTimestamp(timestamp.toString());
 
-  // the remaining time of the unlock cycle
-  const now = moment();
-  const remainingTime = moment.duration(
-    unlockTime.valueOf() - now.valueOf(),
-    "milliseconds",
-  );
+  useEffect(() => {
+    const interval = setInterval(() => {
+      // the remaining time of the unlock cycle
+      const now = moment();
+      const remainingTime = moment.duration(
+        unlockTime.valueOf() - now.valueOf(),
+        "milliseconds",
+      );
+      setRemainingTime(remainingTime);
 
-  // the time already passed in the unlock cycle
-  const alreadyPassedTime = moment.duration(
-    now.valueOf() - startTime.valueOf(),
-    "milliseconds",
-  );
+      // the time already passed in the unlock cycle
+      const alreadyPassedTime = moment.duration(
+        now.valueOf() - startTime.valueOf(),
+        "milliseconds",
+      );
+
+      const percentage =
+        (alreadyPassedTime.asMilliseconds() /
+          (unlockTime.valueOf() - startTime.valueOf())) *
+        100;
+      setPercentageComplete(percentage);
+
+      const remainingTimeInMS = remainingTime?.asMilliseconds();
+
+      if (remainingTimeInMS !== undefined && remainingTimeInMS <= 0) {
+        window.location.reload();
```

Comment:
why reload? 

sorry for all this partial review, seems like I keep missing things =)

Reply:
we had reports on the validators page's epoch interval that it passes 100% and time remaining becomes negative if we don't refresh the page and its over 2 hours epoch time. seems like time interval we used [epoch implementation](https://github.com/aptos-labs/explorer/blob/20db87898930bcba5e7de32b53ebc2ac8cd512e1/src/pages/Validators/Components/Epoch.tsx#L20) will keep updating every min but not aware of the boundary. so i decide to use window reload to force reloading the window when time's up so that we don't see negative time countdown.

<img width="192" alt="Screenshot 2023-02-02 at 5 06 55 PM" src="https://user-images.githubusercontent.com/121921928/216486585-5ffbaef8-6bba-4a1e-98c4-e468c68fc422.png">

Reply:
is there anything specific we need to show when it passes 100%? because we can
```
if (remainingTimeInMS !== undefined && remainingTimeInMS <= 0) {
      // update component to display something we want
      clearInterval(interval);
}
```

Reply:
when it passes 100%, it will reset to another 30 days countdown (30 day is the staking cycle). so the ideal situation would be: when remaining time reaching 0, we keep computing the remaining time with the presumption that unlock time will be updated to 30 days from now. are you suggesting we re-compute remainingTime in the code block like:

```
if (remainingTimeInMS !== undefined && remainingTimeInMS <= 0) {
      const remainingTime = moment.duration(
        unlockTime.valueOf() - now.valueOf(),
        "milliseconds",
      );
      clearInterval(interval);
}
```

alternatively, if we don't do anything in the code block, but just clear the interval, like this [code pointer](https://github.com/aptos-labs/explorer/blob/20db87898930bcba5e7de32b53ebc2ac8cd512e1/src/pages/Validators/Components/Epoch.tsx#L53-L55) is doing, this is why people had issues like [WP-289](https://linear.app/aptoslabs/issue/WP-289/epoch-progress-bar-goes-over-100percent-and-time-remaining-becomes)?